zerver/views/users.py

```diff
@@ -134,7 +133,9 @@ def avatar(request: HttpRequest, user_profile: UserProfile,
         if is_email:
             avatar_user_profile = get_user_including_cross_realm(email_or_id, realm)
         else:
-            avatar_user_profile = get_user_by_id_in_realm_including_cross_realm(int(email_or_id), realm)
+            avatar_user_profile = access_user_by_id(user_profile, int(email_or_id), allow_bots=True,
+                                                    allow_cross_realm=True, allow_deactivated=True,
+                                                    read_only=True)
         # If there is a valid user account passed in, use its avatar
         url = avatar_url(avatar_user_profile, medium=medium)
     except UserProfile.DoesNotExist:
```

Comment:
@timabbott Even in `zerver/views/users.py` , here `UserProfile.DoesNotExist` is being handled externally.
So when there's an invalid `email_or_id` passed to `access_user_by_id` it gets handled inside itself and thus raising jsonable error , whereas it should have been caught by the `try...except` here and handled according to the `avatar` function.